assayist/processor/main_analyzer.py

```diff
@@ -42,6 +42,23 @@ def _read_and_save_buildroots(self):
                 for artifact in self._buildroot_to_artifact[buildroot_id]:
                     artifact.buildroot_artifacts.connect(rpm)
 
+    @staticmethod
+    def _extract_component_name_and_namespace(pull):
+        """Extract component name and namespace from image pull metadata.
+
+        :param str pull: pull metadata
+        :return: component name and namespace
+        :rtype: tuple
+        """
+        pull = pull.split('/', 1)[1]  # strip leading builder hostname
+        pull = pull.rsplit('@', 1)[0]  # strip trailing commit identifier
+
+        cnamespace, _, cname = pull.rpartition('/')
+        if not cnamespace:
+            cnamespace = 'redhat'
```

Comment:
Is this to deal with top-level names like 'rhel7'? I think we should probably have cnamespace being '' in that case.

Reply:
Yes, see e.g. build 792771. Is empty string better than `redhat` in that case?

Reply:
I think so.

Reply:
Ack, removed.